src/librustdoc/html/markdown.rs

```diff
@@ -1034,12 +1034,99 @@ impl MarkdownSummaryLine<'_> {
     }
 }
 
+/// Renders a subset of Markdown in the first paragraph of the provided Markdown.
+///
+/// - Headings, links, and formatting are stripped.
+/// - Inline code **is** rendered.
+/// - HTML and code blocks are ignored.
+///
+/// Returns a tuple of the rendered HTML string and whether the output was shortened
+/// due to the provided `length_limit`.
+fn render_markdown_summary_with_limit(md: &str, length_limit: Option<u16>) -> (String, bool) {
```

Comment:
Can you either:
- Move these files from `html/` to `render/`
- Remove `render_` from the start

Having this called 'render' without being in the render module is very confusing.